### PR TITLE
Add analystics backend to the agent

### PR DESCRIPTION
**Pull request by jasonLaster**
There's one simple "start" event now that ships the main version.

On every new event, we'll include the origin and session_id so we can
track events. A lot of this is still new, so i'm not sure which events
we'll want to gather and how we'll do analytics and what not.

### CODE REVIEW

File: extension/js/agent/actions/analytics.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      event_name: eventName,
+      session_id: session_id,
+      url: window.location.origin
+    }, data);
+
+    saveRecord(event, eventData);
+  }
+
+  Agent.startAnalytics = function() {
+    Parse.initialize("ZMtYo1w9R7U8FcTX4NPSlCCTTYl41PRhEz19yREC", "fjfnW6iqB9KwtRh6yBwuQrLlYPYgwR3reEC4KkNH");
```
Comment (jasonLaster) [inline extension/js/agent/actions/analytics.js]: Is this secure?

Comment (samccone) [inline extension/js/agent/actions/analytics.js]: yeah the client stuff is

File: extension/js/agent/actions/analytics.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+;(function(Agent) {
+
+  var Parse = Agent.Parse;
+  var session_id = Agent.randomString(10);
+
+
+  var saveRecord = function(object, data) {
+    object.save(data, {
```
Comment (samccone) [inline extension/js/agent/actions/analytics.js]: these return promises which are nice

File: extension/js/agent/actions/analytics.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      }
+    });
+  }
+
+  Agent.markEvent = function(eventName, data) {
+    var Event = Parse.Object.extend("Event");
+    var event = new Event();
+
+    var eventData = _.extend({
+      event_name: eventName,
```
Comment (samccone) [inline extension/js/agent/actions/analytics.js]: just pass this in .. just like a new Backbone.Model

File: extension/js/agent/actions/analytics.js (hunk 1 above)
Comment (ianmstew) [inline extension/js/agent/actions/analytics.js]: @samccone what happens if I call Parse.initialize with those hashes from a random app?

File: extension/js/agent/actions/analytics.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      session_id: session_id,
```
Comment (ianmstew) [inline extension/js/agent/actions/analytics.js]: What are your plans for session_id?

File: extension/js/agent/actions/analytics.js (hunk 1 above)
Comment (samccone) [inline extension/js/agent/actions/analytics.js]: you have access unless the app is setup to use roles

Comment (ianmstew) [inline extension/js/agent/actions/analytics.js]: okay, and roles involve an authorized user.  makes sense.

Comment (samccone) [inline extension/js/agent/actions/analytics.js]: :+1:

File: extension/js/agent/actions/analytics.js (hunk 4 above)
Comment (jasonLaster) [inline extension/js/agent/actions/analytics.js]: not sure, probably usage statistics. i'd like a unique id for the user/browser to say how many ppl use the app

**Pull request closed without merging**